src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs

```diff
@@ -27,9 +27,10 @@ static PageHandler CreatePageHandler(IPopup virtualView)
 			view.SetBinding(BindingContextProperty, new Binding { Source = virtualView, Path = BindingContextProperty.PropertyName });
 			var contentPage = new ContentPage
 			{
-				Content = view,
-				Parent = virtualView.Parent as Element
+				Content = view
 			};
+			var parent = virtualView.Parent as Element;
+			parent?.AddLogicalChild(contentPage);
```

Comment:
@pictos can this just be changed to 

```C#
			var parent = (Element)virtualView.Parent;
			parent.AddLogicalChild(contentPage);
```

To be consistent with other platforms? Or are there scenarios where this might actually be null?

Reply:
I just updated the other platforms to also be null safe. At first I thought that the `GetCurrentPage` extension method could never return null, which is why I didn't do that originally, but looking closer at its implementation it seems like there are some scenarios where it might - it's at least hard to guarantee that it doesn't. So now it's consistent, with the null safety check.

Reply:
If the MainPage is null, what will be the state of a Maui application? I mean, is that a correct behavior?

Reply:
If the page is null, the behavior will be the same as it was before my change - the popup likely won't show at all, which arguably seems preferable to introducing an NRE in a scenario that didn't previously NRE. In practice, I don't know if it will ever be null - unlikely, but it's hard to say for certain. @PureWeen may know better. Here's the code for `GetCurrentPage` for instance (which does a `#nullable disable` at the top, unfortunately): https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Platform/PageExtensions.cs#L9. If the shell `CurrentItem` is a `ShellSectionController` and `ShellSectionController.PresentedPage` is null, then `GetCurrentPage` will be null, but I'm not sure if this happens in practice. In any case, I just didn't want to introduce an NRE when there wasn't one there before.

Reply:
@BretJohnson, it's fine to not introduce a NRE... From my point of view, Maui should throw it if this state is invalid. It's fine just not show a popup. Thanks for the references (:

Reply:
I think popups can only exist on pages right? so if `GetCurrentPage` is null it seems like something has gone really wrong :-)